internal/pkg/daemon/enricher/impl.go

```diff
@@ -108,13 +112,61 @@ func (d *defaultImpl) NewForConfig(
 }
 
 func (d *defaultImpl) ListPods(
-	c *kubernetes.Clientset, nodeName string,
+	ctx context.Context, c kubernetes.Interface, nodeName string,
 ) (*v1.PodList, error) {
-	return c.CoreV1().Pods("").List(context.TODO(), metav1.ListOptions{
+	return c.CoreV1().Pods("").List(ctx, metav1.ListOptions{
 		FieldSelector: "spec.nodeName=" + nodeName,
 	})
 }
 
+func (d *defaultImpl) LabelProblematicPod(
+	ctx context.Context, c kubernetes.Interface, containerName string, pod *v1.Pod, l logr.Logger,
+) {
+	// verify if we need to label or if the label is already there
+	if labels := pod.GetLabels(); labels != nil {
+		if _, ok := labels[problematicContainerLabelKey]; ok {
+			return
+		}
+	}
+
+	containerRetryBackoff := wait.Backoff{
+		Duration: backoffDuration,
+		Factor:   backoffFactor,
+		Steps:    backoffSteps,
+	}
+
+	if err := util.RetryEx(
+		&containerRetryBackoff,
+		func() (retryErr error) {
+			return d.doLabelProblematicPodPod(ctx, c, pod)
+		},
+		func(inErr error) bool {
+			return !kerrors.IsNotFound(inErr)
+		},
+	); err != nil {
+		l.Error(
+			err, "unable to patch container to mark it as problematic",
+			"pod.Namespace", pod.GetNamespace(), "pod.Name", pod.GetName(),
+			"containerName", containerName,
+		)
+	}
+}
```

Comment:
I would add that method to the enricher, since it looks like we only have `doLabelProblematicPodPod` to mock.